src/container.rs

```diff
@@ -504,14 +504,19 @@ impl ContainerListOptionsBuilder {
         &mut self,
         filters: Vec<ContainerFilter>,
     ) -> &mut Self {
-        let mut param = HashMap::new();
+        let mut param: HashMap<&str, Vec<String>> = HashMap::new();
         for f in filters {
-            match f {
-                ContainerFilter::ExitCode(c) => param.insert("exit", vec![c.to_string()]),
-                ContainerFilter::Status(s) => param.insert("status", vec![s]),
-                ContainerFilter::LabelName(n) => param.insert("label", vec![n]),
-                ContainerFilter::Label(n, v) => param.insert("label", vec![format!("{}={}", n, v)]),
+            let (key, value) = match f {
+                ContainerFilter::ExitCode(c) => ("exited", c.to_string()),
+                ContainerFilter::Status(s) => ("status", s),
+                ContainerFilter::LabelName(n) => ("label", n),
+                ContainerFilter::Label(n, v) => ("label", format!("{}={}", n, v)),
             };
+            if let Some(values) = param.get_mut(key) {
+                values.push(value);
+            } else {
+                param.insert(key, vec![value]);
+            }
```

Comment:
This could be refactored to something like 

```
param.entry(key).and_modify(|values| values.push(value)).or_insert(key, vec![value]);
```

I guess? Can you try and if yes, push it as additional commit (no squash please, so the history reflects the development properly).

Reply:
Nice!